### PR TITLE
Add SE3.exp_map() and SE3.log_map()

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: theseus/geometry/se3.py [hunk 1]
```diff
@@ -97,10 +97,138 @@ def _hat_matrix_check(matrix: torch.Tensor):
 
     @staticmethod
     def exp_map(tangent_vector: torch.Tensor) -> LieGroup:
-        raise NotImplementedError
+        if tangent_vector.ndim != 2 or tangent_vector.shape[1] != 6:
+            raise ValueError("Tangent vectors of SE(3) can only be 6-D vectors.")
+
+        NEAR_ZERO_EPS = 5e-3
```
Comment (mhmukadam) [inline theseus/geometry/se3.py:103]: We should consolidate all the EPS, otherwise these spread around could make debugging messy.

File: theseus/geometry/tests/test_se3.py [hunk 2]
```diff
@@ -3,20 +3,122 @@
...
+        tangent_vector_ang *= torch.rand(batch_size, 1).double() * 2 * np.pi - np.pi
+        tangent_vector_lin = torch.randn(batch_size, 3).double()
+        tangent_vector = torch.cat([tangent_vector_lin, tangent_vector_ang], dim=1)
+
+        check_exp_map(tangent_vector, th.SE3)
+
+    for batch_size in [1, 20, 100]:
+        tangent_vector_ang = torch.rand(batch_size, 3).double() - 0.5
+        tangent_vector_ang /= tangent_vector_ang.norm(dim=1, keepdim=True)
+        tangent_vector_ang *= 1e-5
```
Comment (mhmukadam) [inline theseus/geometry/tests/test_se3.py:43]: Since all these for loops differ only on this line, maybe leave comments about why we are testing on these sets of values.

**Review by mhmukadam — APPROVED** [review]

File: theseus/geometry/se3.py (hunk 1 above)
Comment (luisenp) [reply theseus/geometry/se3.py:103]: This is part of #70

File: theseus/geometry/tests/test_se3.py (hunk 2 above)
Comment (luisenp) [reply theseus/geometry/tests/test_se3.py:43]: If I understand correctly, this is to test different corner cases (e.g., near zero tangent vectors). If so, we should also add similar tests for SE2 and SO2 (in a new PR). But agree that adding some small comments here would be useful.

Comment (fantaosha) [reply theseus/geometry/tests/test_se3.py:43]: Yeah, as @luisenp said,  `1e-5` and `3e-3` will implement different methods to calculation the `exp_map` and `log_map`.

Comment (fantaosha) [reply theseus/geometry/tests/test_se3.py:43]: I will add some comments about these tests.

Comment (fantaosha) [reply theseus/geometry/tests/test_se3.py:44]: Comments added in the new commit.
